### PR TITLE
select_trains() and split_trains() methods for detector data classes

### LINKED ISSUE #255: split_trains method for xtdf detector classes
Something I noticed while working on code for FXE - it would be useful to be able to split the `LPD1M` object into chunks, like we can with `DataCollection` and `KeyData`. In particular, a `frames_per_part` option would be useful when you want to limit how much data a chunk contains without needing to know the number of frames per train. My idea would still round it to a whole number of trains in each chunk, though.

In a similar vein, `KeyData.split_trains` could have an option like `bytes_per_part`. :thinking:

### PR DESCRIPTION
**Pull request by takluyver**
`DataCollection` and `KeyData` both have methods to select a subset of data by trains, and to split the data along the trains dimension. Various examples would be simpler if the classes for multi-module detector data offered the same API. At present, you have to split your `DataCollection` and call e.g. `AGIPD1M(chunk)` on each piece. This API would let you call `AGIPD1M(run)` up front, and then split the result, which would make various examples more straightforward.

TBD: A `frames_per_part` option for splitting? This would let you split based on a maximum number of frames in a chunk, rather than a maximum number of trains, which would make it easier to write code that stays within a memory budget if the number of frames per train changes.

Closes #255

### CODE REVIEW

**Comment by tmichela** [comment]
Maybe worth adding the additional methods to the reference doc, else LGTM

**Comment by takluyver** [comment]
I've added the `frames_per_part=` parameter, which I hope will be useful for memory budgeting.

The splitting with this is somewhat cruder, because we don't assume that there's a consistent number of frames per train, so the chunks won't necessarily be balanced. We could have a cleverer implementation when frames-per-train is constant, but having two ways to do the same thing is extra complexity.

File: extra_data/tests/test_components.py [hunk 1]
```diff
@@ -265,6 +265,64 @@ def test_get_dask_array_jungfrau(mock_jungfrau_run):
...
+    assert [len(p.train_ids) for p in parts] == [1] * 20
+
+    # trains_per_part cuts off before frames_per_part
+    parts = list(det.split_trains(trains_per_part=3, frames_per_part=1024))
+    assert [len(p.train_ids) for p in parts] == ([3] * 6) + [2]
+
+    # trains_per_part cuts off before frames_per_part
+    parts = list(det.split_trains(trains_per_part=3, frames_per_part=1024))
+    assert [len(p.train_ids) for p in parts] == ([3] * 6) + [2]
+
```
Comment (tmichela) [inline extra_data/tests/test_components.py:307]:
```suggestion
    # trains_per_part cuts off before frames_per_part
    parts = list(det.split_trains(trains_per_part=3, frames_per_part=1024))
    assert [len(p.train_ids) for p in parts] == ([3] * 6) + [2]

```

**Review by tmichela — APPROVED** [review]
LGTM

**Pull request merged**